BunnymodXT/modules/HwDLL.cpp

```diff
@@ -3940,6 +3954,422 @@ struct HwDLL::Cmd_BXT_Show_Bullets_Enemy_Clear
 	}
 };
 
+struct HwDLL::Cmd_BXT_Split
+{
+	USAGE("Usage: bxt_split <name>\n Tells BunnySplit to split by the specified map name, and prints to console the current time. Mostly intended to be used in scripts, or in custom triggers (these may not be valid in runs, check the rules for your game/category).\n");
+
+	// TODO: it should be possible to split without a map/split name, like when you right-click on LiveSplit and click on Split,
+	// but I don't think BunnySplit supports this
+
+	static void handler(const char* name)
+	{
+		Vector speed;
+		Vector origin;
+		const auto& player = HwDLL::GetInstance().GetPlayerEdict();
+		if (player)
+		{
+			speed = player->v.velocity;
+			origin = player->v.origin;
+		}
+		const auto& time = CustomHud::GetTime();
+
+		Splits::Split fake_split;
+		fake_split.set_name(name);
+		fake_split.set_time(time);
+		fake_split.set_speed(speed);
+		fake_split.set_origin(origin);
+
+		Splits::PrintSplitCompletion(fake_split);
+		Interprocess::WriteMapChange(time, name);
+	}
+};
+
+struct HwDLL::Cmd_BXT_Split_On
+{
+	USAGE("Usage: bxt_split_on <target_name> [map_name]\n Tells BunnySplit to split when activating an entity by the specified name, and prints to console the current time.\n");
+
+	static void handler(const char* targetname)
+	{
+		Splits::splits.emplace_back(std::string(targetname));
+	}
+
+	static void handler(const char* targetname, const char* map_name)
+	{
+		Splits::splits.emplace_back(std::string(targetname), std::string(map_name));
+	}
+};
+
+struct HwDLL::Cmd_BXT_Splits_Add
+{
+	USAGE("Usage: bxt_splits_add <x1> <y1> <z1> <x2> <y2> <z2> [map_name] [name]\n Adds a split trigger in a form of axis-aligned cuboid with opposite corners at coordinates (x1, y1, z1) and (x2, y2, z2).\n");
+
+	static void handler(float x1, float y1, float z1, float x2, float y2, float z2)
+	{
+		Splits::splits.emplace_back(Vector(x1, y1, z1), Vector(x2, y2, z2));
+	}
+
+	static void handler(float x1, float y1, float z1, float x2, float y2, float z2, const char* map_name)
+	{
+		Splits::splits.emplace_back(Vector(x1, y1, z1), Vector(x2, y2, z2), std::string(map_name));
+	}
+
+	static void handler(float x1, float y1, float z1, float x2, float y2, float z2, const char* map_name, const char* split_name)
+	{
+		Splits::splits.emplace_back(Vector(x1, y1, z1), Vector(x2, y2, z2), std::string(map_name), std::string(split_name));
+	}
+};
+
+struct HwDLL::Cmd_BXT_Splits_Clear
+{
+	NO_USAGE()
+
+	static void handler()
+	{
+		Splits::splits.clear();
+	}
+};
+
+struct HwDLL::Cmd_BXT_Splits_Delete
+{
+	USAGE("Usage: bxt_splits_delete [id]\n Deletes the last placed split trigger.\n If an id is given, deletes the split trigger with the given id.\n");
+
+	static void handler()
+	{
+		if (Splits::splits.empty()) {
+			HwDLL::GetInstance().ORIG_Con_Printf("You haven't placed any split triggers.\n");
```

Comment:
```suggestion
			HwDLL::GetInstance().ORIG_Con_Printf("You haven't added any splits.\n");
```
Since there aren't only triggers?